x-pack/plugins/infra/public/components/asset_details/tabs/metadata/utils.ts

```diff
@@ -9,108 +9,65 @@ import type { InfraMetadata } from '../../../../../common/http_api';
 
 export interface Field {
   name: string;
-  value: string | string[] | undefined;
+  value?: string | string[];
+}
+
+export interface FieldsByCategoryValue {
+  value?: string | boolean | string[] | { [key: string]: string };
+}
+export interface FieldsByCategory {
+  [key: string]: FieldsByCategoryValue['value'];
 }
 
 export const getAllFields = (metadata: InfraMetadata | null) => {
   if (!metadata?.info) return [];
-  return prune([
-    {
-      name: 'host.architecture',
-      value: metadata.info.host?.architecture,
-    },
-    {
-      name: 'host.hostname',
-      value: metadata.info.host?.name,
-    },
-    {
-      name: 'host.id',
-      value: metadata.info.host?.id,
-    },
-    {
-      name: 'host.ip',
-      value: metadata.info.host?.ip,
-    },
-    {
-      name: 'host.mac',
-      value: metadata.info.host?.mac,
-    },
-    {
-      name: 'host.name',
-      value: metadata.info.host?.name,
-    },
-    {
-      name: 'host.os.build',
-      value: metadata.info.host?.os?.build,
-    },
-    {
-      name: 'host.os.family',
-      value: metadata.info.host?.os?.family,
-    },
-    {
-      name: 'host.os.name',
-      value: metadata.info.host?.os?.name,
-    },
-    {
-      name: 'host.os.kernel',
-      value: metadata.info.host?.os?.kernel,
-    },
-    {
-      name: 'host.os.platform',
-      value: metadata.info.host?.os?.platform,
-    },
-    {
-      name: 'host.os.version',
-      value: metadata.info.host?.os?.version,
-    },
-    {
-      name: 'cloud.account.id',
-      value: metadata.info.cloud?.account?.id,
-    },
-    {
-      name: 'cloud.account.name',
-      value: metadata.info.cloud?.account?.name,
-    },
-    {
-      name: 'cloud.availability_zone',
-      value: metadata.info.cloud?.availability_zone,
-    },
-    {
-      name: 'cloud.instance.id',
-      value: metadata.info.cloud?.instance?.id,
-    },
-    {
-      name: 'cloud.instance.name',
-      value: metadata.info.cloud?.instance?.name,
-    },
-    {
-      name: 'cloud.machine.type',
-      value: metadata.info.cloud?.machine?.type,
-    },
-    {
-      name: 'cloud.provider',
-      value: metadata.info.cloud?.provider,
-    },
-    {
-      name: 'cloud.region',
-      value: metadata.info.cloud?.region,
-    },
-    {
-      name: 'agent.id',
-      value: metadata.info.agent?.id,
-    },
-    {
-      name: 'agent.version',
-      value: metadata.info.agent?.version,
-    },
-    {
-      name: 'agent.policy',
-      value: metadata.info.agent?.policy,
-    },
-  ]);
+  const tempPropertyMap = new Map<string, FieldsByCategoryValue['value']>();
```

Comment:
Q: why do we need this?

Reply:
Good point, I had a different approach at the beginning - there is no need for it, thanks for spotting that. We can just use your suggestion to have the `const value = fieldsByCategory[property];` and use the value directly. I removed it 👍